### ISSUE TITLE
Auth flag

### ISSUE THREAD

**Issue by Christilut** [vkarpov15/run-rs#19]
So my development MongoDB is exposed because it is on a Lightsail VPS and I'd like to secure it with the `--auth` flag. 
I tried adding it myself, seemed like a small feature but I'm getting some errors in the underlying topology manager when adding the auth flag.

**Comment by Christilut** [comment]
Well I want to load real data in there and instead of putting effort into anonymizing the data I'd rather secure it. There's bots out there that just go after every IP address and test every default port, especially mongo since the default mongo installations have auth disabled.
I changed my ports of course but IP whitelisting is proving difficult because my server is on heroku which uses dynamic IP's.

**Comment by Christilut** [comment]
But honestly, I had runrs running for a few days and it worked fine but lately I've been getting strange errors and the primary clusters seems to go down repeatedly after 12+ hours of uptime. After messing with it (and other ways of getting a replicaset with auth running) I decided it costs me too much time to get it working and that an M10 instance on Atlas is maybe the best option after all.

**Comment by vkarpov15** [comment]
It sounds like your case is not what run-rs is designed for. Our goal is to make local dev and testing as easy as possible. As soon as you start worrying about uptime and securing customer data, you should look into MongoDB Atlas. I'll keep this open to track adding auth support though, that will be useful.

**Comment by Christilut** [comment]
Yeah you're right. I was kind of forced to do something like this because of [this issue](https://stackoverflow.com/questions/53254164/mongodb-atlas-error-while-performing-transaction-on-multiple-collections-code-8)

**Comment by vkarpov15** [comment]
Thanks for the heads up, I've never seen that Atlas issue before. I opened up a Mongoose issue to look into it.

**Comment by ashisherc** [comment]
would love to have auth flag, when running a dev cluster on a cloud VM

**Comment by rattrayalex** [comment]
It looks like Prisma requires auth for local dev: https://www.prisma.io/docs/concepts/database-connectors/mongodb

lack of this flag seems to make this project unusable for that…
